utilities/backupable/backupable_db.cc

```diff
@@ -320,29 +326,40 @@ class BackupEngineImpl : public BackupEngine {
     return GetSharedChecksumDirRel() + "/" + (tmp ? "." : "") + file +
            (tmp ? ".tmp" : "");
   }
-  inline bool UseSessionId(const std::string& sid) const {
-    return GetTableNamingOption() == kOptionalChecksumAndDbSessionId &&
-           !sid.empty();
+  inline bool UseLegacyNaming(const std::string& sid) const {
+    return GetNamingNoFlags() ==
+               BackupableDBOptions::kLegacyCrc32cAndFileSize ||
+           sid.empty();
+  }
+  inline bool UseInterimNaming(const std::string& sid) const {
+    // The indicator of SST file from early internal 6.12 release
+    // is a '-' in the DB session id. DB session id was made more
```

Comment:
Nice, I would not have guessed we could further exploit the session ID to prevent duplication in another filename schema change!